### PR TITLE
Update openshift-gitops permissions to fix pipeline-service syncing

### PR DESCRIPTION
**Pull request by Mo3m3n**
- use `argocd.argoproj.io/managed-by: openshift-gitops` label to give openshift-gitops the permission to deploy to a given namespace. This is a syntactic sugar replacing the rolebinding of openshit-gitops to the admin role in the specified namespace.
- Add missing permissions for pipelines-as-code (servicemonitors, validatingwebhookconfigurations)

### CODE REVIEW

**Review by Roming22 — COMMENTED** [review]
lgtm, but you'll need to fix the YAML formatting. Remember that you should be able to run `developer/hack/ci.sh` to validate your change before submitting the PR.

File: operator/gitops/argocd/pipeline-service/pipelines-as-code/allow-argocd.yaml [hunk 1]
```diff
@@ -1,42 +1,39 @@
...
+      - admissionregistration.k8s.io
     resources:
-      - customresourcedefinitions
+      - validatingwebhookconfigurations
+    verbs:
+      - create
+  - apiGroups:
+      - monitoring.coreos.com
+    resources:
+      - servicemonitors
```
Comment (bnallapeta) [inline operator/gitops/argocd/pipeline-service/pipelines-as-code/allow-argocd.yaml:25]: Why are we providing permission to ArgoCD for collecting metrics? Could you help me understand where its collecting and displaying those metrics?

Comment (Mo3m3n) [reply operator/gitops/argocd/pipeline-service/pipelines-as-code/allow-argocd.yaml:25]: pipelines-as-code [requires](https://github.com/openshift-pipelines/pipelines-as-code/releases/download/v0.14.2/release.yaml) the creation of a servicemonitor, so we are providing argoCD with the permissions to create that.

**Pull request merged**